### PR TITLE
'Python_Implementation_of_Rabin_Karp'

### LINKED ISSUE #2019: Python Implementation of Rabin Karp Algorithm
@jainaman224 
There is no Python implementation for the Rabin Karp algorithm. I would like to work on this issue for GSSoC. Please assign me with this issue.

### PR DESCRIPTION
**Pull request by codermuskanj**
Fixes #2019 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Python Implementation of Rabin Karp Algorithm

#### Languages Used:
- Python

#### Files Added:
-    ###Rabin_Karp.py

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by codermuskanj** [comment]
@jainaman224 @somya-kapoor @raghav-dalmia I have implemented Rabin Karp Algo in Python as it was assigned to me. Kindly go through it.

File: Rabin_Karp/Rabin_Karp.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    for i in range (patlen - 1) :
+        val = (val * AN) % PN       # PN is a prime number
+
+    for i in range (patlen) :
+        hpat = (AN * hpat + ord(pattern[i])) % PN
+        htxt = (AN * htxt + ord(text[i])) % PN
+
+    for i in range (txtlen - patlen + 1) :
+        if  hpat == htxt :
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:32]: you kept two spaces after 'if', please remove a space

File: Rabin_Karp/Rabin_Karp.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    for i in range (patlen) :
+        hpat = (AN * hpat + ord(pattern[i])) % PN
+        htxt = (AN * htxt + ord(text[i])) % PN
+
+    for i in range (txtlen - patlen + 1) :
+        if  hpat == htxt :       
+            for j in range (patlen) :
+                 if text[i + j] != pattern[j]:
+                     break
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:35]: correct the indentation

File: Rabin_Karp/Rabin_Karp.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            j += 1
+            if j == patlen :
+               print( "Found at index starting from " + str(i))
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:38]: correct the indentation

File: Rabin_Karp/Rabin_Karp.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        # calculate the new position including i + m   
+        if i < txtlen - patlen :
+            htxt = (AN * (htxt -  ord(text[i])*val) + ord(text[i + patlen])) % PN
+            
+            if htxt < 0 :
+                 htxt = htxt + PN
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:45]: correct the indentation

File: Rabin_Karp/Rabin_Karp.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        # calculate the new position including i + m   
+        if i < txtlen - patlen :
+            htxt = (AN * (htxt -  ord(text[i])*val) + ord(text[i + patlen])) % PN
+            
+            if htxt < 0 :
+                 htxt = htxt + PN           
+
+# Driver program
+text = "XWINGO XOR WINGO"   
+pattern = "WINGO"
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:49]: can you take dynamic inputs?

File: Rabin_Karp/Rabin_Karp.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+#        WINGO
+#       101
+#OUTPUT:-Found at index starting from 1
+#       Found at index starting from 11
+
+    
+
+
+    
+
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:64]: remove these lines

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by codermuskanj** [comment]
yes @somya-kapoor

**Comment by codermuskanj** [comment]
@asha15 I have done the required changes. Kindly label my Pr so that it could be reflected in my score in Gssoc.

File: Rabin_Karp/Rabin_Karp.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    for i in range (patlen) :
+        hpat = (AN * hpat + ord(pattern[i])) % PN
+        htxt = (AN * htxt + ord(text[i])) % PN
+
+    for i in range (txtlen - patlen + 1) :
+        if hpat == htxt :       
+            for j in range (patlen) :
+                if text[i + j] != pattern[j]:
+                   break
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:35]: correct the indentation

File: Rabin_Karp/Rabin_Karp.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            j += 1
+            if j == patlen :
+               print( "Found at index starting from " + str(i))
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:38]: correct the indentation

File: Rabin_Karp/Rabin_Karp.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+        # calculate the new position including i + m   
+        if i < txtlen - patlen :
+           htxt = (AN * (htxt -  ord(text[i])*val) + ord(text[i + patlen])) % PN
+            
+           if htxt < 0 :
+              htxt = htxt + PN
```
Comment (asha15) [inline Rabin_Karp/Rabin_Karp.py:45]: correct the indentation 
keep space around operators

**Comment by codermuskanj** [comment]
@somya-kapoor @jainaman224 @asha15 I have done the required changes. Kindly provide label to my PR so that it can be reflected in my Gssoc score when merged.

**Pull request merged**